decode_test.go

```diff
@@ -8366,6 +8366,72 @@ func TestUnmarshalFieldNameByteString(t *testing.T) {
 	}
 }
 
+func TestDecModeInvalidReturnTypeForEmptyInterface(t *testing.T) {
+	for _, tc := range []struct {
+		name         string
+		opts         DecOptions
+		wantErrorMsg string
+	}{
+		{
+			name:         "below range of valid modes",
+			opts:         DecOptions{ReturnTypeForEmptyIface: -1},
+			wantErrorMsg: "cbor: invalid ReturnTypeForEmptyIface -1",
+		},
+		{
+			name:         "above range of valid modes",
+			opts:         DecOptions{ReturnTypeForEmptyIface: 101},
+			wantErrorMsg: "cbor: invalid ReturnTypeForEmptyIface 101",
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			_, err := tc.opts.DecMode()
+			if err == nil {
+				t.Errorf("DecMode() didn't return an error")
+			} else if err.Error() != tc.wantErrorMsg {
+				t.Errorf("DecMode() returned error %q, want %q", err.Error(), tc.wantErrorMsg)
+			}
+		})
+	}
+}
+
+func TestUnmarshalTaggedDataToEmptyInterfaceWithReturnType(t *testing.T) {
+	for _, tc := range []struct {
+		name string
+		opts DecOptions
+		in   []byte
+		want interface{}
+	}{
+		{
+			name: "default to value of type Tag",
+			opts: DecOptions{},
+			in:   hexDecode("d8ff00"),
+			want: Tag{},
+		},
+		{
+			name: "Tag's content",
+			opts: DecOptions{ReturnTypeForEmptyIface: ReturnTagContentForEmptyInterface},
+			in:   hexDecode("d8ff00"),
+			want: uint64(0),
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			dm, err := tc.opts.DecMode()
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			var got interface{}
+			if err := dm.Unmarshal(tc.in, &got); err != nil {
+				t.Errorf("unexpected error: %v", err)
+			}
+
+			if reflect.TypeOf(tc.want) != reflect.TypeOf(got) {
+				t.Errorf("got %s, want %s", reflect.TypeOf(got), reflect.TypeOf(tc.want))
+			}
```

Comment:
We should probably verify the decoded value here.  

Also, it would be good to add tests to make sure:
- tags with tag number 0, 1, 2, 3 are not affected by the new option
- tags with tag number registered by `TagSet` are not affected by the new option

Reply:
Thanks for the suggestion. I've added more tests covering the suggested cases.